### PR TITLE
adoptopenjdk: add livecheck

### PR DESCRIPTION
**Pull request by SeekingMeaning**
Now works with https://github.com/Homebrew/brew/pull/10376 merged

### CODE REVIEW

File: Casks/adoptopenjdk.rb [hunk 1]
```diff
@@ -8,6 +8,18 @@
   name "AdoptOpenJDK Java Development Kit"
   homepage "https://adoptopenjdk.net/"
 
+  major_version = version.major
```
Comment (vitorgalvao) [inline Casks/adoptopenjdk.rb:11]: Why the variable? Won’t `version.major` work inside the `livecheck`?

Comment (reitermarkus) [reply Casks/adoptopenjdk.rb:11]: It does.

Comment (SeekingMeaning) [reply Casks/adoptopenjdk.rb]: It doesn't...?

Comment (SeekingMeaning) [reply Casks/adoptopenjdk.rb]:
```console
$ brew livecheck --cask adoptopenjdk
adoptopenjdk : 15,36 ==> 5,
```

`version.major` evaluates to `nil`

Comment (reitermarkus) [reply Casks/adoptopenjdk.rb]: It definitely should work inside `livecheck`:

https://github.com/Homebrew/brew/blob/e989a0fb011e0f2f8b61cd195c4703c71a4b4b7f/Library/Homebrew/livecheck.rb#L111-L112

Comment (SeekingMeaning) [reply Casks/adoptopenjdk.rb]: Not sure why it's not working, then

**Review by reitermarkus — DISMISSED** [review]

**Comment by miccal** [comment]
@SeekingMeaning and @reitermarkus I attempted to resolve a merge conflict, but may have inadvertently un-done something -- apologies.

**Comment by reitermarkus** [comment]
It's still correct. The `Version` handling needs to be fixed in `brew` so that `"15,36" < "15.0.1,9.1"`.

**Comment by miccal** [comment]
Thank you @SeekingMeaning.

**Review by miccal — APPROVED** [review]

**Pull request merged**